src/LinAlg/hiopMatrixRajaSparseTriplet.cpp

```diff
@@ -459,6 +459,57 @@ double hiopMatrixRajaSparseTriplet::max_abs_value()
   return maxv;
 }
 
+void hiopMatrixRajaSparseTriplet::row_max_abs_value(hiopVector& ret_vec)
+{
+  assert(ret_vec.get_size() == nrows_);
+  
+  auto& vec = dynamic_cast<hiopVectorRajaPar&>(ret_vec);
+  vec.setToZero();
+  
+  double* vd = vec.local_data();
+  
+  auto iRow = this->iRow_;
+  auto values = this->values_;
+ 
+  for (int i = 0; i < nrows_; ++i)
+  {
+    RAJA::ReduceMax<hiop_raja_reduce, double> norm(0.0);
+    RAJA::forall<hiop_raja_exec>(RAJA::RangeSegment(0, nnz_),
+    RAJA_LAMBDA(RAJA::Index_type itnz)
+    {
+      const int row_id = iRow[itnz];
+      if(i==row_id) {
+        norm.max(std::abs(values[itnz]));
+      }
+    });
+
+    vd[i] = norm.get();
+  }
```

Comment:
I don't think this is a sensible implementation, and it is not efficient at all.
I wonder if there is a RAJA array of type RAJA::ReduceMax, and then we can use it to recode the norm of each row.
Then we can use the following code without the outer for loop.
    RAJA::ReduceArray<RAJA::ReduceMax<hiop_raja_reduce, double>> norm_row(nrows_, 0.0); 
    RAJA::forall<hiop_raja_exec>(RAJA::RangeSegment(0, nnz_),
    RAJA_LAMBDA(RAJA::Index_type itnz)
    {
      const int row_id = iRow[itnz];
      norm_row[row_id].max(std::abs(values[itnz]));
    });
@cnpetra

Reply:
I don't think that's available. I think this can be implemented efficiently with RAJA::forall and list segments (will require no reduce ops). Will pm you...